src/main/java/it/gov/pagopa/pu/workflow/wf/debtposition/custom/fine/wfreductionexpiration/FineReductionOptionExpirationWFImpl.java

```diff
@@ -22,11 +30,26 @@ public class FineReductionOptionExpirationWFImpl implements FineReductionOptionE
    */
   @Override
   public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
-    // TODO https://pagopa.atlassian.net/browse/P4ADEV-2494
+    DebtPositionFineWfConfig wfConfig = applicationContext.getBean(DebtPositionFineWfConfig.class);
+
+    debtPositionFineReductionOptionExpirationActivity = wfConfig.buildDebtPositionFineReductionOptionExpirationActivityStub();
+    invokeSyncDebtPositionActivity = wfConfig.buildInvokeSyncDebtPositionActivity();
   }
 
-  public String handleFineReductionExpiration(Long debtPositionId, PaymentEventRequestDTO paymentEventRequestDTO, FineWfExecutionConfig executionParams) {
-    // TODO https://pagopa.atlassian.net/browse/P4ADEV-2494
-    return "";
+  public String expireFineReduction(Long debtPositionId, FineWfExecutionConfig executionParams) {
+    log.info("Handling fine reduction expiration for debtPositionId: {}, executionParams: {}", debtPositionId, executionParams);
+    DebtPositionDTO debtPositionDTO = debtPositionFineReductionOptionExpirationActivity.handleFineReductionExpiration(debtPositionId);
+
+    if (debtPositionDTO == null){
+      log.warn("DebtPositionDTO not found for debtPositionId: {}", debtPositionId);
+      return null;
+    }
+
+    // TODO replace IO placeholders https://pagopa.atlassian.net/browse/P4ADEV-2599
+    log.info("Mapped FineWfExecutionConfig: {} to GenericWfExecutionConfig", executionParams);
+    GenericWfExecutionConfig wfExecutionConfig = FineWfExecutionConfigMapper.mapReductionExpired(executionParams);
```

Comment:
Thus this should be renamed to avoid name conflict
```suggestion
    GenericWfExecutionConfig genericWfExecutionConfig = FineWfExecutionConfigMapper.mapReductionExpired(executionParams);
```